lib/rule-context.js

```diff
@@ -72,42 +72,29 @@ var PASSTHROUGHS = [
  * @param {object} ecmaFeatures The ecmaFeatures settings passed from the config file.
  */
 function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures) {
+    // public.
+    this.id = ruleId;
+    this.options = options;
+    this.settings = settings;
+    this.ecmaFeatures = ecmaFeatures;
```

Comment:
Hmm, `ecmaFeatures` still can be modified, and 3-4% slower. Is it needed?
If it's needed, should I treat `options` and `settings` with the same way?

※ `RuleTester` is freezing `ecmaFeatures`, `settings`, and `options` deeply.

```js
context.ecmaFeatures.__proto__.modules = true;
```

| `this.ecmaFeatures = ???`                    | Time   |
|----------------------------------------------|-------:|
| `ecmaFeatures`                               | 3004ms |
| `Object.freeze(Object.create(ecmaFeatures))` | 3112ms |
| `assign({}, ecmaFeatures)`                   | 3095ms |
| `Object.freeze(assign({}, ecmaFeatures))`    | 3115ms |

If it treats `settings` and `options` as well, then 9-10% slower. 
`RuleContext` constructor is called many times (rules ✖ files), so this has big impact to performance.